### DISCUSSION TITLE
Self-Host Share Threshold

### DISCUSSION THREAD

**Discussion by ealeksandrov7** [Web3Auth/Web3Auth#849] [Help]
Hey guys. I understand how you generate new shares when using the Self-Host flow, but let's say I'm using it together with the new Node SDK:
By using the Node SDK I can get a PK to use as 'first share' so then my wallet access becomes 1/1.
If I add a security question/answer - I get 1/2 and so on.
How do I increase the threshold instead of just adding additional shares for 1/n security.
How do I go from threshold of '1' to a threshold of 'n'?

Thanks for the new awesome SDK btw!

**Comment by ealeksandrov7** [comment]
Actually can we use the private key generated from the Node.js SDK as a postboxKey when initializing a new ThresholdKey?

**Reply by himanshuchawla009** [reply]
No, it is not suggested to do that. The key returned from node.js sdk is the final user's key, its not postbox key. We can look in to adding a method to return postbox key as well. Please raise it as a feature request on the repo.

**Reply by ealeksandrov7** [reply]
Okay, got it - thanks! And do you think it's a problem if I generate my own "postbox key" and use it when initializing a service provider.
What I can see from my tests is that there's no problem with that. After adding the desired shares (security question/answer and a local share) I am able to consistently generate the same pk/seed phrase. 
Of course that postbox key is stored in a db and used to initialize the service provider in following attempts to login.

**Reply by YZhenY** [reply]
theres a security risk here with using a naive setup for the service-provider, in particular with Web3Auth we don't support a flow using an external service provider at this point in time
